extension/templates/devTools/activity/graph.js

```diff
@@ -0,0 +1,90 @@
+define(['d3', 'util/Radio'], function(d3, Radio) {
+  
+  return {
+
+    formatData: function(data) {
+
+      var rectHeight = 20;
+      var startX = Infinity;
+      var endX = -Infinity;
+
+      _.each(data, function(activity) {
+
+        activity = activity.attributes;
+
+        startX = activity.startTime < startX ? activity.startTime : startX;
+        endX = activity.endTime > endX ? activity.endTime : endX;
+
+        activity.position = {
+          "dx": activity.endTime - activity.startTime,
+          "dy": rectHeight,
+          "x": activity.startTime,
+          "y": rectHeight * activity.depth
+        };
+
+      });
+
+      return {
+        data: data,
+        startX: startX,
+        endX: endX
+      };
+
+    },
+
+    displayGraph: function (formattedData, startX, endX) {
+
+      var activityGraph = d3.select('div.activity-graph');
+
+      var margin = 2;
+      var width = +activityGraph.style('width').split('px')[0];
+      var height = 100;
+
+      var x = d3.scale.linear()
+        .domain([startX, endX])
+        .range([0, width]);
+
+      var y = d3.scale.linear()
+        .domain([0, 100])
+        .range([0, height]);
+
+      var color = d3.scale.category20();
+
+      var svg = activityGraph.append("svg")
+        .attr("width", width - (2 * margin))
+        .attr("height", height - (2 * margin));
+
+      rect = svg.selectAll('rect')
+          .data(formattedData)
+        .enter().append("rect")
+          .attr("x", function(d) { return x(d.attributes.position.x); })
+          .attr("y", function(d) { return y(d.attributes.position.y); })
+          .attr("width", function(d) { return 5 * d.attributes.position.dx; })
+          .attr("height", function(d) { return d.attributes.position.dy; })
+          .attr("fill", function(d) { return color(d.attributes.eventName); })
+          .on("click", onClick)
+          .on("mouseover", onMouseover);
+
+      function onClick(d) {
+
+        var eventId = d.attributes.eventId;
+        Radio.command('activity', 'click:graph', eventId);
+
+      }
```

Comment:
I'm tempted to suggest moving all this code into the view, let me show you what I mean:

```js
ActivityGraph = View.extend({

  formatData: function() {},
  displayGraph: function() {
      // stuff
      svg.selectAll('rect').data(formattedData).enter().append('rect').attrs().on("click", _.bind(this, this.onClick)
  },

  onClick: function() {
     var eventId = d.attributes.eventId;
     var activityModel = this.activityCollection.findByEventId(eventId);
     Radio.command('activity', 'show:info', activityModel);
  },
  onMouseover: function() {
  }
});